File: task_queue/Worker.java
import com.rabbitmq.client.*;
import java.io.IOException;

public class Worker {

    private static final String TASK_QUEUE_NAME = "task_queue";

    public static void main(String[] argv) throws Exception {
        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost("localhost");
        final Connection connection = factory.newConnection();
        final Channel channel = connection.createChannel();

        //open queue
        boolean durable = true;
        channel.queueDeclare(TASK_QUEUE_NAME, durable, false, false, null); //durable=true to make the queue persistent
        channel.basicQos(1); //a worker gets new task only after it finishes its current one;
                             //make channel to accept only consumer that has one unack-ed message at a time

        //start consumer
        final Consumer consumer = new DefaultConsumer(channel) {
            @Override
            public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) throws IOException {
                String message = new String(body, "UTF-8");

                System.out.println(" [x] Received '" + message + "'");
                try {
                    doWork(message);
                } finally {
                    System.out.println(" [x] Done");
                    channel.basicAck(envelope.getDeliveryTag(), false);
                }
            }
        };
        System.out.println(" [*] Waiting for messages. To exit press CTRL+C");
        boolean autoAck = false; // acknowledgment is covered below
        channel.basicConsume(TASK_QUEUE_NAME, autoAck, consumer);
    }

    private static void doWork(String task) {
        /*emulate loong task as sleeping 1 second for each dot '.'*/

        for (char ch : task.toCharArray()) {
            if (ch == '.') {
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException _ignored) {
                    Thread.currentThread().interrupt();
                }
            }
        }
    }

}

File: task_queue/worker.py
#!/usr/bin/env python
import pika
import time

connection = pika.BlockingConnection(pika.ConnectionParameters(host='localhost'))
channel = connection.channel()

channel.queue_declare(queue='task_queue', durable=True) #durable=True to keep queue persistent


def callback(ch, method, properties, body):
  #process each dot '.' in 1 second i.e. emulate a loong task
  print(" [x] Received %r" % body)
  time.sleep(body.count(b'.'))
  print(" [x] Done")

  #send an ack to acknowledge rabbitmq server that the task/message is done i.e. it can be deleted/no re-queue for it
  ch.basic_ack(delivery_tag=method.delivery_tag)


channel.basic_qos(prefetch_count=1) #a worker only get new task only after it finishes its current one
channel.basic_consume(callback, queue='task_queue')

print(' [*] Waiting for messages. To exit press CTRL+C')
channel.start_consuming()
